src/SourceBuild/tarball/patches/roslyn/0001-lift-version-of-Microsoft.CodeAnalysis.Common-depend.patch

```diff
@@ -0,0 +1,25 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Logan Bussell <36081148+lbussell@users.noreply.github.com>
+Date: Thu, 21 Oct 2021 23:15:23 +0000
+Subject: [PATCH] lift version of Microsoft.CodeAnalysis.Common dependencies to
+ previously source built versions
+
+---
+ .../Source/CSharpSyntaxGenerator/CSharpSyntaxGenerator.csproj | 4 ++++
+ 1 file changed, 4 insertions(+)
+
+diff --git a/src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/CSharpSyntaxGenerator.csproj b/src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/CSharpSyntaxGenerator.csproj
+index 7eae8afa275..a50822a6aeb 100644
+--- a/src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/CSharpSyntaxGenerator.csproj
++++ b/src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/CSharpSyntaxGenerator.csproj
+@@ -27,5 +27,9 @@
+     <Compile Include="..\..\..\..\..\Compilers\CSharp\Portable\Syntax\SyntaxKindFacts.cs" Link="Grammar\SyntaxKindFacts.cs" Condition="'$(TargetFramework)' != 'netstandard2.0'" />
+ 
+     <PackageReference Include="Microsoft.CodeAnalysis.Common" Version="$(SourceGeneratorMicrosoftCodeAnalysisVersion)" PrivateAssets="all" Condition="'$(TargetFramework)' == 'netstandard2.0'" />
++    <PackageReference Include="System.Collections.Immutable" Version="$(SourceBuildLiftedSystemCollectionsImmutableVersion)" PrivateAssets="all" Condition="'$(TargetFramework)' == 'netstandard2.0'" />
++    <PackageReference Include="System.Reflection.Metadata" Version="$(SourceBuildLiftedSystemReflectionMetadataVersion)" PrivateAssets="all" Condition="'$(TargetFramework)' == 'netstandard2.0'" />
++    <PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="$(SourceBuildLiftedSystemRuntimeCompilerServicesUnsafeVersion)" PrivateAssets="all" Condition="'$(TargetFramework)' == 'netstandard2.0'" />
++    <PackageReference Include="System.Text.Encoding.CodePages" Version="$(SourceBuildLiftedSystemTextEncodingCodePagesVersion)" PrivateAssets="all" Condition="'$(TargetFramework)' == 'netstandard2.0'" />
```

Comment:
A few notes about this patch, nothing to do pre-merge IMO:
* If upstream were to accept the patch, these package references would probably need to go into an `ItemGroup` that only takes effect during source-build, because `SourceBuildLiftedSystemCollectionsImmutableVersion` is only defined during the tarball build and I think `Version=""` would cause a build failure.
* I'm not sure upstream would accept it: they would probably prefer putting `Microsoft.CodeAnalysis.Common` in SBRP. But, this doesn't matter: my understanding is that we don't have time to put `Microsoft.CodeAnalysis.Common` in SBRP.
* It's enticing to simply turn the `PackageReference Include="Microsoft.CodeAnalysis.Common"` into a ProjectReference to remove the dependency on previouslysourcebuilt. The reason this doesn't work is that `Microsoft.CodeAnalysis.Common` already depends on `CSharpSyntaxGenerator` as a project reference. The Roslyn repo is using this PackageReference to break a very tight dependency cycle.